### PR TITLE
Add iteration_utilities recipe

### PR DESCRIPTION
**Pull request by MSeifert04**
I tried to follow the guidelines and compared with some existing recipes. Let me know if I did something wrong.

### CODE REVIEW

**Comment by MSeifert04** [comment]
Could anyone point me to instructions how to setup the tests for the `meta.yml` if the `tests` folder is included in the tarball but not included in the installed package?

File: recipes/iteration_utilities/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+test:
+  requires:
+    - pytest
+  imports:
+    - iteration_utilities
+
+about:
+  home: https://github.com/MSeifert04/iteration_utilities
+  license: Apache v.2
```
Comment (jakirkham) [inline recipes/iteration_utilities/meta.yaml]: Please make these tweaks.

```yaml
  license: Apache 2.0
  license_family: Apache
```

**Comment by MSeifert04** [comment]
@jakirkham Thank you for your comments. 👍 I've updated the recipe accordingly.

File: recipes/iteration_utilities/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  license_file: LICENSE.rst
+  summary: 'Itertools recipes for Python 2/3 and more.'
+  doc_url: http://iteration-utilities.rtfd.io
+  dev_url: https://github.com/MSeifert04/iteration_utilities
+
+extra:
+  recipe-maintainers:
+    - MSeifert04
+    - LukeSkywalker
```
Comment (MSeifert04) [inline recipes/iteration_utilities/meta.yaml]: Yes, that was an oversight. I'm the only maintainer.

File: recipes/iteration_utilities/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  number: 0
+  features:
+  - vc9  # [win and py27]
+  - vc10  # [win and py34]
+  - vc14  # [win and py35]
+  skip: True  # [osx]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/iteration_utilities/meta.yaml]: Sorry I didn't realize this was compiling code before. Could we please add `toolchain`?

Comment (MSeifert04) [inline recipes/iteration_utilities/meta.yaml]: 👍

**Comment by jakirkham** [comment]
Thanks @MSeifert04 .

**Pull request merged**